internal/databases/sqlserver/blob/server.go

```diff
@@ -603,36 +604,52 @@ func (bs *Server) HandleBlobPut(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	blobSizeStr := req.Header.Get("Content-Length")
-	blobSize, err := strconv.ParseUint(blobSizeStr, 10, 64)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
 	garbage := idx.Clear()
-	if blobSize > 0 {
-		name := idx.PutBlock("data", blobSize)
-		encryptedReader := internal.CompressAndEncrypt(req.Body, bs.compressor, bs.crypter)
-		err := folder.PutObject(name, encryptedReader)
-		req.Body.Close()
-		if err != nil {
-			bs.returnError(w, req, err)
-			return
+	if blobSizeStr != "0" {
+		// positive Content-Length or empty (if chunked encoding)
+		defer req.Body.Close()
+		const blockSize = 4 * 1024 * 1024
+		buf := make([]byte, blockSize)
+		xblocklist := &XBlockListIn{Blocks: []XBlockIn{}}
+		for i, last := 0, false; !last; i++ {
+			n, err := io.ReadFull(req.Body, buf)
+			if err != nil {
+				if err == io.EOF || err == io.ErrUnexpectedEOF {
+					last = true
+				} else {
+					bs.returnError(w, req, err)
+					return
+				}
+			}
+			if n == 0 {
+				continue
+			}
+			id := fmt.Sprintf("data_%05d", i)
+			name := idx.PutBlock(id, uint64(n))
+			encryptedReader := internal.CompressAndEncrypt(bytes.NewReader(buf[:n]), bs.compressor, bs.crypter)
+			err = folder.PutObject(name, encryptedReader)
```

Comment:
I am just curious about why do you split blob into 4Mb chunks.

Reply:
SQLServer uses 4Mb blocks by itself. We mimic.